_posts/2021-01-06-how-to-update-to-cis-13.adoc

```diff
@@ -0,0 +1,124 @@
+
+=== Step 1: Update references to the Gruntwork Infrastructure as Code Library
+
+In order to update to the CIS AWS Foundations Benchmark v1.3.0, you need to update your references to the Gruntwork
+Infrastructure as Code Library to use a compatible version. We (Gruntwork) have reviewed and updated all the modules in the
+library for compatibility with the new version of the Benchmark. As a customer, you need to update to
+the proper versions of the Gruntwork library to pick up the fixes/changes that were made to be compatible. Refer to
+https://gruntwork.io/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library/#updating[the
+"Updating" section of "How to use the Gruntwork Infrastructure as Code Library"] for instructions on how to update the
+versions in your code.
+
+For the vast majority of the repos, the only change that will be necessary is a version number bump, but several repos
+require more extensive code changes and state migrations. To upgrade without downtime and data loss, **you MUST follow
+the migration instructions in the release notes in each repo to know what changes need to be made to update to the new
+version.**
+
+[.exceptional]
+IMPORTANT: Gruntwork follows
+https://gruntwork.io/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library/#versioning[semantic
+versioning]. For any pre-1.0 modules, this means that version updates to the minor version are considered backwards
+incompatible releases for any version updates prior to 1.0.0 release. Make sure to read the release notes for the
+relevant modules any time you are updating minor versions! Note that you will want to read the release notes for each
+minor version that is updated (e.g., if you are going from v0.5.x to v0.9.x, you will want to read the notes for v0.6.0,
+v0.7.0, v0.8.0, and v0.9.0 to get the full list of backwards incompatible updates).
+
+The following table provides a summary of all the relevant Gruntwork AWS modules and the respective versions that are
+compatible with CIS AWS 1.3.0:
+
+[[compatibility_table]]
+[cols="1,1h,1"]
+|===
+|Gruntwork Repo |Minimum version with CIS AWS 1.3.0 support |Corresponding CIS AWS 1.3.0 recommendations
+
+|module-security
+|https://github.com/gruntwork-io/module-security/releases/tag/v0.44.5[v0.44.5]
+|1.20, 2.1.1, 2.1.2, 3.10, 3.11
+
+|terraform-aws-monitoring
+|https://github.com/gruntwork-io/terraform-aws-monitoring/releases/tag/v0.24.0[v0.24.0]
+|1.20, 2.1.1, 2.1.2
+
+|package-zookeeper
+|https://github.com/gruntwork-io/package-zookeeper/releases/tag/v0.8.0[v0.8.0]
+|1.20, 2.1.1, 2.1.2
+
+|terraform-aws-vpc
+|https://github.com/gruntwork-io/terraform-aws-vpc/releases/tag/v0.12.0[v0.12.0]
+|1.20, 2.1.1, 2.1.2
+
+|package-openvpn
+|https://github.com/gruntwork-io/package-openvpn/releases/tag/v0.13.0[v0.13.0]
+|1.20, 2.1.1, 2.1.2
+
+|cis-compliance-aws
+|https://github.com/gruntwork-io/cis-compliance-aws/releases/tag/v0.9.2[v0.9.2]
+|4.15, 3.10, 3.11
+
+|===
+
+
+=== Step 2: Manual steps
```

Comment:
I think there's quite a bit more to do than this... Many of the requirements require deploying new modules that customers would not have been running at all previously, so bumping version numbers alone is not enough. Going off the list in [this Jira epic](https://gruntwork.atlassian.net/browse/IAC-1604):

1. You need to deploy the IAM access analyzer module.
2. You have to request increased NACL limits before deploying the new NACL stuff.
3. You have to deploy the new `cleanup-expired-certs` module.
4. You have to deploy the new `ebs-encryption-multi-region` module.

There may be others, so make sure all the people that worked on these parts take a careful look!

Reply:
Yeah, definitely, that's why this PR is still WIP. Everyone who worked on the CIS upgrade is supposed to add their bits to this guide.

Reply:
I'm wondering 🤔  besides the tickets we had as part of this epic in Jira (https://gruntwork.atlassian.net/browse/IAC-1604), do we have another way of gathering data when people have done work towards CIS 1.3 outside of the project itself (e.g. the `ebs-encryption-multi-region`)? I tried searching slack, but couldn't really dig out anything else really

Reply:
Hm, not really...

@bwhaley When you went through the new requirements for v1.3, did you compare it to just v1.2, and create tickets for each new item in v1.3? Or did you compare it to _what we already had at Gruntwork_, and only create tickets for the new items we'd have to create at Gruntwork? If it's the latter, then there are new modules or features we added since we released our 1.2 offering that we need to explicitly tell customers about in this upgrade guide?

Reply:
I was curious to have a play around with this bug, and I have to say it's very annoying and unfortunately my lack of experience with Jekyll and playing around didn't really help!

Anyway, I thought I'd share a few things I looked into, and hopefully it could be a very simple fix when combined with more knowledgeable people in how this works! 

https://jekyllrb.com/docs/liquid/filters/
https://shopify.github.io/liquid/filters/escape/

Basically, as far as I could gather, none of the `;`, `,`, `.`, `:` characters work, so I assume it's some sort of escaping of them that needs to happen!

Reply:
I looked at the Benchmark v1.3.0, compared each control to 1.2, and opened a ticket for each difference. I do not think there are any modules or steps that we need to cover that are not in this epic. For instance, the EBS encryption module was already underway when I put the epic together, and it was released before this project began, but it's still mentioned here in a ticket. Hope that makes sense - let me know if it doesn't!

Reply:
Ahaa! That makes sense, @bwhaley :) Thanks for confirming - I can see that the EBS is indeed under the Jira epic, but last time I must have missed it or something, so my bad here! 

Ok, so in this case we're all good and confident all that's on Jira is indeed the correct list of changes we need to mention 🙂